tutorials/introductory/animation_tutorial.py

```diff
@@ -0,0 +1,224 @@
+# parameter to determine the length of the animation. The *interval* parameter
+# is used to determine time in milliseconds between drawing of two frames.
+# Animating using `.FuncAnimation` would usually follow the following
+# structure:
+#
+# - Plot the initial figure, including all the required artists. Save all the
+#   artists in variables so that they can be updated later on during the
+#   animation.
+# - Create an animation function that updates the data in each artist to
+#   generate the new frame at each function call.
+# - Create a `.FuncAnimation` object with the `.Figure` and the animation
+#   function, along with the keyword arguments that determine the animation
+#   properties.
+# - Use `.animation.Animation.save` or `.pyplot.show` to save or show the
+#   animation.
+#
+# The update function uses the ``set_*`` function for different artists to
+# modify the data. The following table shows a few plotting methods, the artist
+# types they return and some methods that can be used to update them.
+#
+# =================  =============================  ===========================
+# Plotting method    Artist                         Set method
+# =================  =============================  ===========================
+# `.Axes.plot`       `.lines.Line2D`                `.lines.Line2D.set_data`
+# `.Axes.scatter`    `.collections.PathCollection`  `.collections.\
+#                                                   PathCollection.set_offsets`
+# `.Axes.imshow`     `.image.AxesImage`             ``AxesImage.set_data``
+# `.Axes.annotate`   `.text.Annotation`             `.text.Annotation.\
+#                                                   update_positions`
+# `.Axes.barh`       `.patches.Rectangle`           `.Rectangle.set_angle`,
+#                                                   `.Rectangle.set_bounds`,
+#                                                   `.Rectangle.set_height`,
+#                                                   `.Rectangle.set_width`,
+#                                                   `.Rectangle.set_x`,
+#                                                   `.Rectangle.set_y`
+#                                                   `.Rectangle.set_xy`
+# `.Axes.fill`       `.patches.Polygon`             `.Polygon.set_xy`
+# `.patches.Circle`  `.patches.Ellipse`             `.Ellipse.set_angle`,
+#                                                   `.Ellipse.set_center`,
+#                                                   `.Ellipse.set_height`,
+#                                                   `.Ellipse.set_width`
+# =================  =============================  ===========================
+#
+# Covering the set methods for all types of artists is beyond the scope of this
+# tutorial but can be found in their respective documentations. An example of
+# such update methods in use for `.Axes.scatter` is as follows.
+
+
+fig, ax = plt.subplots()
+t = np.linspace(-4, 4, 400)
```

Comment:
Can we tweak the range a little bit so the end matches the beginning instead of overlapping slightly?

Reply:
How about now? Does this look better?

Reply:
Yea, it looks like it's 2π long though, so could we do `-np.pi` to `np.pi`?

Reply:
Yeah, let me check that.

Reply:
Nice observation, thanks! Done.